lib/config/rule-validator.js

```diff
@@ -35,16 +35,30 @@ function findRuleDefinition(ruleId, config) {
         pluginName = ruleIdParts.join("/");
     }
 
-    if (!config.plugins || !config.plugins[pluginName]) {
-        throw new TypeError(`Key "rules": Key "${ruleId}": Could not find plugin "${pluginName}".`);
-    }
+    if (config.plugins) {
 
-    if (!config.plugins[pluginName].rules || !config.plugins[pluginName].rules[ruleName]) {
-        throw new TypeError(`Key "rules": Key "${ruleId}": Could not find "${ruleName}" in plugin "${pluginName}".`);
-    }
+        const plugin = config.plugins[pluginName];
+
+        // first check for exact rule match
+        if (plugin && plugin.rules && plugin.rules[ruleName]) {
+            return config.plugins[pluginName].rules[ruleName];
+        }
+
+        let errorMessage = `Key "rules": Key "${ruleId}": Could not find "${ruleName}" in plugin "${pluginName}".`;
 
-    return config.plugins[pluginName].rules[ruleName];
+        // otherwise, let's see if we can find the rule name elsewhere
+        for (const [otherPluginName, otherPlugin] of Object.entries(config.plugins)) {
+            if (otherPlugin.rules && otherPlugin.rules[ruleName]) {
+                errorMessage += ` Did you mean "${otherPluginName}/${ruleName}"?`;
```

Comment:
This change looks good, but these messages are generally confusing when they relate to core rules because of `@/`, for example:

```
TypeError: Key "rules": Key "foo/no-undef": Could not find "no-undef" in plugin "foo". Did you mean "@/no-undef"?
```

Reply:
That's a UX thing I'll work on along the way.